gallery_dl/text.py

```diff
@@ -51,6 +51,21 @@ def slugify(value):
     return re.sub(r"[-\s]+", "-", value).strip("-_")
 
 
+def sanitize_for_filename(string):
+    """Removes characters from a string that would be illegal to have in
+    a filename
+
+    This function is similar to slugify(), except it retains more
+    characters (notably characters such as # and @).
+
+    Note that the length of the string is not capped!
+
+    Inspiration:
+    https://stackoverflow.com/a/71199182
+    """
+    return re.sub(r"[/\\?%*:|\"<>\x7F\x00-\x1F]", " ", str(string))
+
+
```

Comment:
This is already handled by [`path-restrict`](https://gdl-org.github.io/docs/configuration.html#extractor-path-restrict) and [`path-remove`](https://gdl-org.github.io/docs/configuration.html#extractor-path-remove) in [path.py](https://github.com/mikf/gallery-dl/blob/master/gallery_dl/path.py).
https://github.com/mikf/gallery-dl/blob/454f766f5ec249de93317610d375e84899f7b35c/gallery_dl/path.py#L92
https://github.com/mikf/gallery-dl/blob/454f766f5ec249de93317610d375e84899f7b35c/gallery_dl/path.py#L106